plugins/schemaorg/custom/src/Extension/Custom.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  Schemaorg.Custom
+ *
+ * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Plugin\Schemaorg\Custom\Extension;
+
+use Joomla\CMS\Event\Plugin\System\Schemaorg\BeforeCompileHeadEvent;
+use Joomla\CMS\Event\Plugin\System\Schemaorg\PrepareSaveEvent;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\CMS\Schemaorg\SchemaorgPluginTrait;
+use Joomla\Event\Priority;
+use Joomla\Event\SubscriberInterface;
+use Joomla\Registry\Registry;
+
+// phpcs:disable PSR1.Files.SideEffects
+\defined('_JEXEC') or die;
+// phpcs:enable PSR1.Files.SideEffects
+
+/**
+ * Schemaorg Plugin
+ *
+ * @since   __DEPLOY_VERSION__
+ */
+final class Custom extends CMSPlugin implements SubscriberInterface
+{
+    use SchemaorgPluginTrait;
+
+    /**
+     * Load the language file on instantiation.
+     *
+     * @var    boolean
+     * @since  __DEPLOY_VERSION__
+     */
+    protected $autoloadLanguage = true;
+
+    /**
+     * The name of the schema form
+     *
+     * @var   string
+     * @since __DEPLOY_VERSION__
+     */
+    protected $pluginName = 'Custom';
+
+    /**
+     * Returns an array of events this subscriber will listen to.
+     *
+     * @return  array
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            'onSchemaPrepareForm'       => 'onSchemaPrepareForm',
+            'onSchemaBeforeCompileHead' => ['onSchemaBeforeCompileHead', Priority::BELOW_NORMAL],
+            'onSchemaPrepareSave'       => 'onSchemaPrepareSave',
+        ];
+    }
+
+    public function onSchemaPrepareSave(PrepareSaveEvent $event): void
+    {
+        $subject = $event->getData();
+
+        if (empty($subject->schemaType) || $subject->schemaType !== 'Custom' || !isset($subject->schema)) {
+            return;
+        }
+
+        try {
+            $schema = new Registry($subject->schema);
+
+            $json = (new Registry($schema->get('json')))->toArray();
+        } catch (\RuntimeException $e) {
+            throw new \RuntimeException('PLG_SCHEMAORG_CUSTOM_JSON_ERROR');
+        }
+
+        if (!isset($json['@context']) || $json['@context'] !== 'https://schema.org' || !isset($json['@type'])) {
```

Comment:
```suggestion
        if (!isset($json['@context']) || !preg_match('#^https://schema.org(/.*)?$#', $json['@context']) || !isset($json['@type'])) {
```

Some available online generators generate the @context url with a trailing slash. 
Since both the Schema.org validator and also the google rich-results test consider the entry here to be valid, we should perhaps not be so strict here either to be more user-friendly?